src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -547,4 +570,47 @@ public Mono<Void> removeModificationsToExclude(UUID nodeUuid, List<UUID> modific
     public void notifyModificationNodeChanged(UUID studyUuid, UUID nodeUuid) {
         emitNodesChanged(studyUuid, List.of(nodeUuid));
     }
+
+    @Transactional
+    public UUID doGetReportUuid(UUID nodeUuid, boolean generateId) {
+        return nodesRepository.findById(nodeUuid).map(n -> repositories.get(n.getType()).getReportUuid(nodeUuid, generateId)).orElse(null);
+    }
+
+    public Mono<UUID> getReportUuid(UUID nodeUuid) {
+        return Mono.fromCallable(() -> self.doGetReportUuid(nodeUuid, true))
+            .switchIfEmpty(Mono.error(new StudyException(ELEMENT_NOT_FOUND)));
+    }
+
+    private void getParentReportUuidsAndNamesFromNode(NodeEntity nodeEntity, boolean nodeOnlyReport, List<Pair<UUID, String>> res) {
```

Comment:
Not optimied !
An another US [https://dev.azure.com/gridsuite/MVP/_backlogs/backlog/MVP%20Team/Features/?workitem=971](url)